### PR TITLE
fixes #7052 - adding pulpnode plugin support

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
@domcleal updated

File: manifests/plugin/pulp.pp [hunk 1]
```diff
@@ -4,27 +4,29 @@
...
 # === Parameters:
 #
-# $group::        group owner of the configuration file
+# $group::            group owner of the configuration file
 #
-# $enabled::      enables/disables the plugin
+# $enabled::          enables/disables the pulp plugin
 #
-# $pulp_url::     pulp url to use
+# $pulpnode_enabled:: enables/disables the pulpnode plugin
```
Comment (domcleal) [inline manifests/plugin/pulp.pp]: `type:boolean` (and above for enabled, actually)

File: manifests/plugin/pulp.pp [hunk 2]
```diff
@@ -4,27 +4,29 @@
...
-
   foreman_proxy::plugin {'pulp':
   } ->
-  file {'/etc/foreman-proxy/settings.d/pulp.yml':
-    ensure  => file,
-    content => template('foreman_proxy/plugin/pulp.yml.erb'),
-    owner   => 'root',
-    group   => $group_real,
-    mode    => '0640',
+  foreman_proxy::settings_file { 'pulp':
```
Comment (ekohl) [inline manifests/plugin/pulp.pp]: I don't know why we didn't push for settings_file more in plugins. Nice catch.

File: manifests/plugin/pulp.pp (hunk 1 above)
Comment (ekohl) [inline manifests/plugin/pulp.pp]: For bonus points you can add ```validate_bool``` in the body as well.

**Comment by jlsherrill** [comment]
@domcleal good catch, didn't even realize there was a type field.   

@ekohl i like bonus points :)

**Comment by domcleal** [comment]
:+1:

**Comment by ekohl** [comment]
Thanks!

**Pull request closed without merging**